### PR TITLE
[MIG] migrate to version 15.0

### PR DESCRIPTION
**Pull request by petrus-v**
I would suggest to **not** merge this branch in master, instead as it's common in odoo community maintain one branch per version. So creating a new branch 15.0 from master that I could address this PR. Is that fine to you ?

I've tried to keep compatibility with previous versions but Odoo code has been refactored so it would results on ugly hack.

This first part makes standard payment method working. lyramulti is not supported by this PR.

I have add unit test method in order to improve quality, it could be nice to add integration test and configure runner but this is outside of the scope of this PR and wonder if you @lyra have already some CI configured ?

### CODE REVIEW

File: payment_lyra/helpers/constants.py [hunk 1]
```diff
@@ -7,108 +7,205 @@
...
+    "ja": "Japanese",
+    "nl": "Dutch",
+    "pl": "Polish",
+    "pt": "Portuguese",
+    "ru": "Russian",
+    "sv": "Swedish",
+    "tr": "Turkish",
 }
 
+# values from https://payzen.io/en-EN/form-payment/reference/vads-payment-cards.html
```
Comment (petrus-v) [inline payment_lyra/helpers/constants.py:49]: this comments: https://github.com/lyra/plugin-odoo/issues/15#issuecomment-1012255533 make me realized that it should use those data https://docs.lyra.com/fr/collect/form-payment/standard-payment/vads-payment-cards.html  instead

File: payment_lyra/helpers/constants.py [hunk 2]
```diff
@@ -7,108 +7,205 @@
...
+    "KEY_PROD": "2222222222222222",
+    "CTX_MODE": "TEST",
+    "SIGN_ALGO": "SHA-256",
+    "LANGUAGE": "en",
+    "GATEWAY_VERSION": "V2",
+    "PLUGIN_VERSION": "2.0.0",
+    "CMS_IDENTIFIER": "Odoo_15",
 }
 
+# https://payzen.io/en-EN/form-payment/reference/vads-language.html
```
Comment (petrus-v) [inline payment_lyra/helpers/constants.py:32]: this comments: https://github.com/lyra/plugin-odoo/issues/15#issuecomment-1012255533 make me realized that it should use those data https://docs.lyra.com/fr/collect/form-payment/standard-payment/vads-language.html  instead

File: payment_lyra/helpers/constants.py [hunk 3]
```diff
@@ -7,108 +7,205 @@
...
+    "UNION_PAY": "Union Pay",
+    "VILLAVERDE": "Villaverde gift card",
+    "VILLAVERDE_SB": "Villaverde gift card - Sandbox mode",
+    "VISA": "Visa",
+    "VISA_ELECTRON": "Visa Electron",
+    "VPAY": "Vpay",
+    "WECHAT": "WeChat",
 }
 
+# https://payzen.io/en-EN/form-payment/reference/vads-currency.html
```
Comment (petrus-v) [inline payment_lyra/helpers/constants.py:138]: this comments: #15 (comment) make me realized that it should use those data https://docs.lyra.com/fr/collect/form-payment/standard-payment/vads-currency.html instead

**Comment by nabil509** [comment]
Hi @petrus-v ,

Your pull request must be proposed to our develop branch. Please modify it and compare to our develop branch. We pushed all our changes and we have now a beta version compatible with Odoo 15.

**Review by hchabane21 — COMMENTED** [review]
The different configurations in the code concern the white label Lyra Collect and not PayZen.

**Comment by petrus-v** [comment]
I've change the base and as expected we have conflicts... have to figure out what's the best todo

**Comment by nabil509** [comment]
Hi @petrus-v,

Thank you for this PR. Sorry, we cannot merge it easily because there many conflicts. But we checked that we already included all your changes. 

You can now download the new version according to your offer here: https://github.com/lyra/plugin-odoo/releases/tag/v2.0.0

**Pull request closed without merging**